### PR TITLE
CONTRIBUTING: fill out the style section

### PR DESCRIPTION
**Pull request by vapier**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -65,3 +65,68 @@ Explain the problem and include additional details to help maintainers reproduce
...
+### Git Commit Messages
+
+Rather than duplicate a ton of great explanations, please read this site:
+http://chris.beams.io/posts/git-commit/
+
+Yes, it's long, but once you read and understand it once, it applies to
+every git project out there, not just GD.
+
+In case that site is broken, here is an archived link:
+http://web.archive.org/web/20160706012209/http://chris.beams.io/posts/git-commit/
```
Comment (cmb69) [inline CONTRIBUTING.md]: Not sure if we should link to the wayback machine.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -65,3 +65,68 @@ Explain the problem and include additional details to help maintainers reproduce
...
+* Tests should be standalone: One test should test one thing only.
+* Tests should be reproducible: Avoid relying on system settings like /etc.
+* Tests must be fast: A good guideline is that they should take <1 second,
+  and must take <5 seconds.
+* Tests must be documented: Just add a comment block to the top of the file
+  with a short description and any existing bugs/URLs.
+
+### Documentation Styleguide
+
+Do everything in markdown. It integrates well with GitHub and is readable even
```
Comment (cmb69) [inline CONTRIBUTING.md]: It appears to be worthwhile to clarify that NaturalDocs documentation doesn't support markdown, but has [its own syntax](http://www.naturaldocs.org/documenting/reference.html#FormattingAndLayout).

**Review by cmb69 — COMMENTED** [review]
Other than that, it looks very good to me.

File: CONTRIBUTING.md (hunk 1 above)
Comment (vapier) [inline CONTRIBUTING.md]: then the fallback is working ? :)

we've had links break before and have gaps before they're updated.  i think people understand at this point it's best effort.

**Comment by vapier** [comment]
i polished the documentation section, but left the wayback link.  ptal!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -65,3 +65,74 @@ Explain the problem and include additional details to help maintainers reproduce
 
 
 This text has been largely inspired by the Atom contributing notes, which amazingly complete
```
Comment (cmb69) [inline CONTRIBUTING.md]: Maybe you want to fix "which amazingly complete" as well (should probably be "which are amazingly complete").

**Review by cmb69 — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md (hunk 3 above)
Comment (vapier) [inline CONTRIBUTING.md]: done!

**Pull request merged**